src/components/faq/FaqPage.tsx

```diff
@@ -17,51 +16,43 @@ import {
+const FAQ_PAGE_QUESTIONS: Record<string, ContentType[]> = {
+  [FaqCategory.Common]: COMMON_QUESTIONS,
+  [FaqCategory.Campaigns]: CAMPAIGN_QUESTIONS,
+  [FaqCategory.Donations]: DONATION_QUESTIONS,
+  [FaqCategory.AttractDonators]: ATTRACTING_DONATORS_QUESTIONS,
+  [FaqCategory.CorporatePartnership]: PARTNERSHIPS_QUESTIONS,
+}
 
-export default function FaqPage({ section }: { section: string }) {
+export default function FaqPage({ section }: { section: FaqCategory }) {
   const { t } = useTranslation()
   const [value, setValue] = useState(section)
+  const [searchKeyword, setSearchKeyword] = useState('')
 
-  if (faqOnHold) {
-    return (
-      <Layout title={t('nav.campaigns.faq')}>
-        <OnHold />
-      </Layout>
-    )
-  }
   return (
     <Layout title={t('nav.campaigns.faq')}>
       {/* <FaqIntro /> */}
+      <FaqSearch onChange={setSearchKeyword} />
       <TabContext value={value}>
         <Stack direction={{ xs: 'column', md: 'row' }}>
           <VerticalTabs setValue={setValue} />
-          <TabPanel value="common-questions" sx={{ p: 0 }}>
-            {COMMON_QUESTIONS.flatMap(({ header, content, visible }) =>
-              visible ? <ExpandableListItem key={header} header={header} content={content} /> : [],
-            )}
-          </TabPanel>
-          <TabPanel value="campaigns" sx={{ p: 0 }}>
-            {CAMPAIGN_QUESTIONS.flatMap(({ header, content, visible }) =>
-              visible ? <ExpandableListItem key={header} header={header} content={content} /> : [],
-            )}
-          </TabPanel>
-          <TabPanel value="donations" sx={{ p: 0 }}>
-            {DONATION_QUESTIONS.flatMap(({ header, content, visible }) =>
-              visible ? <ExpandableListItem key={header} header={header} content={content} /> : [],
-            )}
-          </TabPanel>
-          <TabPanel value="attracting-donators" sx={{ p: 0 }}>
-            {ATTRACTING_DONATORS_QUESTIONS.flatMap(({ header, content, visible }) =>
-              visible ? <ExpandableListItem key={header} header={header} content={content} /> : [],
-            )}
-          </TabPanel>
-          <TabPanel value="corporate-partnership" sx={{ p: 0 }}>
-            {PARTNERSHIPS_QUESTIONS.flatMap(({ header, content, visible }) =>
-              visible ? <ExpandableListItem key={header} header={header} content={content} /> : [],
-            )}
-          </TabPanel>
+          {Object.keys(FAQ_PAGE_QUESTIONS).map((categoryKey) => {
+            return (
+              <TabPanel key={categoryKey} value={categoryKey} sx={{ p: 0 }}>
+                {FAQ_PAGE_QUESTIONS[categoryKey]
+                  .filter(filterFaqQuestionByVisibility)
+                  .filter((question) => filterFaqQuestionBySearchKeyword(question, searchKeyword))
+                  .flatMap(({ header, content }) => (
+                    <ExpandableListItem key={header} header={header} content={content} />
+                  ))}
+              </TabPanel>
```

Comment:
Maybe you can automatically expand the list item once a match has been find. And if it matches only a single ExpandableListItem. Because I imagine it won't be good to open multiple ones